internal/controller/nodetemplate_controller_test.go

```diff
@@ -0,0 +1,800 @@
+			By("creating Node")
+			nodes := []corev1.Node{
+				newNode("node1").withLabel(map[string]string{"node-role.kubernetes.io/worker": "true"}).withSpareTaint().build(),
+				newNode("node2").withLabel(map[string]string{"node-role.kubernetes.io/worker": "true"}).build(),
+			}
+			for _, node := range nodes {
+				err := k8sClient.Create(ctx, &node)
+				Expect(err).ToNot(HaveOccurred())
+			}
+
+			By("checking Node status")
+			Eventually(func(g Gomega) {
+				nodeList := &corev1.NodeList{}
+				err := k8sClient.List(ctx, nodeList)
+				g.Expect(err).ToNot(HaveOccurred())
+				for _, node := range nodeList.Items {
+					if node.Name == "node1" {
+						g.Expect(node.Labels).To(HaveKeyWithValue("node-role.kubernetes.io/spare", "true"))
+					} else if node.Name == "node2" {
+						g.Expect(node.Labels).ToNot(HaveKey("node-role.kubernetes.io/spare"))
+					}
+				}
+			}).WithTimeout(10 * time.Second).WithPolling(500 * time.Millisecond).Should(Succeed())
+		})
+
+		It("should successfully allocate the nodes", func() {
+			By("creating Node")
+			nodes := []corev1.Node{
+				newNode("node1").withLabel(map[string]string{"node-role.kubernetes.io/worker": "true"}).withSpareTaint().build(),
+				newNode("node2").withLabel(map[string]string{"node-role.kubernetes.io/worker": "true"}).withSpareTaint().build(),
+				newNode("node3").withLabel(map[string]string{"node-role.kubernetes.io/worker": "true"}).withSpareTaint().build(),
+			}
+			for _, node := range nodes {
+				err := k8sClient.Create(ctx, &node)
+				Expect(err).ToNot(HaveOccurred())
+			}
+			By("creating NodeTemplate")
+			nodeTemplate := &nyallocatorv1.NodeTemplate{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test",
+				},
+				Spec: nyallocatorv1.NodeTemplateSpec{
+					DryRun:   false,
+					Priority: 1,
+					Nodes:    2,
+					Selector: &metav1.LabelSelector{
+						MatchLabels: map[string]string{
+							"node-role.kubernetes.io/worker": "true",
+						},
+					},
+					Template: nyallocatorv1.NodeConfiguration{
+						Metadata: nyallocatorv1.NodeConfigurationMetadata{
+							Labels: map[string]string{
+								"test-label": "foo",
+							},
+							Annotations: map[string]string{
+								"test-annotation": "bar",
+							},
+						},
+						Spec: nyallocatorv1.NodeConfigurationSpec{
+							Taints: []corev1.Taint{
+								{
+									Key:    "test-taint",
+									Value:  "baz",
+									Effect: corev1.TaintEffectNoSchedule,
+								},
+							},
+						},
+					},
+				},
+			}
+			err := k8sClient.Create(ctx, nodeTemplate)
+			Expect(err).ToNot(HaveOccurred())
+
+			By("checking NodeTemplate status")
+			checkNodeTemplateStatus("test", true)
+
+			By("checking node is properly configured")
+			n := &corev1.NodeList{}
+			err = k8sClient.List(ctx, n, client.MatchingLabels{"nyallocator.cybozu.io/node-template": "test"})
+			Expect(err).ToNot(HaveOccurred())
+			Expect(n.Items).To(HaveLen(2), "should have created 2 nodes")
```

Comment:
I don't think it's `create`.
```suggestion
			Expect(n.Items).To(HaveLen(2), "should have allocated 2 nodes")
```